### PR TITLE
Add COMPAS framework recipe

### PR DESCRIPTION
**Pull request by gonzalocasas**
A python framework for collaboration and research in architecture, engineering and digital fabrication.

### CODE REVIEW

**Comment by gonzalocasas** [comment]
ping @conda-forge/core 😃  
It would be extreeeemly awesome if you, folks, could take a look at this and merge... soon-ish if possible ;)

File: recipes/compas/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    digital fabrication. It is developed by the Block Research
+    Group (https://block.arch.ethz.ch) of ETH Zurich (http://www.ethz.ch),
+    with the support of the National Centre for Competence in
+    Research on "Digital Fabrication" (http://www.dfab.ch).
+  doc_url: https://compas-dev.github.io
+  dev_url: https://github.com/compas-dev/compas
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
```
Comment (synapticarbors) [inline recipes/compas/meta.yaml]: Delete comments

File: recipes/compas/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - brgcode
```
Comment (synapticarbors) [inline recipes/compas/meta.yaml]: Has the listed co-maintainer signed off on being listed? If not, please ping them here and get permission.

File: recipes/compas/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+
+requirements:
+  host:
```
Comment (synapticarbors) [inline recipes/compas/meta.yaml]: I think only `python`, `pip` and `setuptools` are needed in the `host` section. If it's not imported in the `setup.py` you don't need it unless it's listed in `setup_requires`

File: recipes/compas/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
```
Comment (synapticarbors) [inline recipes/compas/meta.yaml]: Can this be `noarch: python`

https://conda-forge.org/docs/meta.html#building-noarch-packages

**Comment by gonzalocasas** [comment]
@synapticarbors thank you so much! I'll reply on-thread to each comment, while CI builds it :)

File: recipes/compas/meta.yaml (hunk 1 above)
Comment (gonzalocasas) [reply recipes/compas/meta.yaml]: fixed

File: recipes/compas/meta.yaml (hunk 2 above)
Comment (gonzalocasas) [reply recipes/compas/meta.yaml]: yes, @brgcode is the main developer of compas, and we work together in the same institute.

File: recipes/compas/meta.yaml (hunk 3 above)
Comment (gonzalocasas) [reply recipes/compas/meta.yaml]: fixed, this was the output of `conda skeleton`.

File: recipes/compas/meta.yaml (hunk 4 above)
Comment (gonzalocasas) [reply recipes/compas/meta.yaml]: Unfortunately not really `noarch`. This framework is basically a cross architecture tool, and chances are we'll add more and more arch-specific stuff in the near future.

**Comment by marcelotrevisani** [comment]
Looks good to me. I'm going to merge this

**Pull request merged**